lib/tokenlist.cpp

```diff
@@ -1557,6 +1557,36 @@ static Token * findAstTop(Token *tok1, const Token *tok2)
     return nullptr;
 }
 
+static Token *skipMethodDeclEnding(Token *tok)
+{
+    if (Token::simpleMatch(tok, ")"))
```

Comment:
I feel this duplicates logic in `Tokenizer::isFunctionHead` however it does not handle all the keywords. If possible I would rather reuse that functionality.

    const Token* ftok = tok->link();
    if (Token* tok2 = isFunctionHead(ftok, ";{"))
        return tok->next() != tok2 ? ; tok2 : nullptr;

It should be ok to move isFunctionHead to TokenList, as far as I see that does not impact dependencies at least.